src/utils/address.ts

```diff
@@ -0,0 +1,61 @@
+import { Address, OfficerFiling } from "@companieshouse/api-sdk-node/dist/services/officer-filing";
+import { CompanyAppointment } from "private-api-sdk-node/dist/services/company-appointments/types";
+
+/**
+ * Determine whether an update has occured for correspondence address in the given officer filing.
+ * @param officerFiling 
+ * @param companyAppointment 
+ * @returns 
+ */
+export const checkCorrespondenceAddressUpdate = (officerFiling: OfficerFiling, companyAppointment: CompanyAppointment): boolean => {
+  if(officerFiling.isServiceAddressSameAsRegisteredOfficeAddress !== companyAppointment.isServiceAddressSameAsRegisteredOfficeAddress){
```

Comment:
Flags could be undefined. We should only compare addresses if officerFiling.isServiceAddressSameAsRegisteredOfficeAddress !== true (and currently the flag is also !== true)?

Reply:
New check added as discussed, however I placed it in the controller rather than the utils to keep the utils cleaner. Would you agree this is appropriate?